Add slack analytics project

MajaCernja opened the pull request:
## Description
This PR adds `slack_analytics` dbt project containing a few models and macros for processing message and reactions data accessible via the public Slack API. 

It also adds 6 tasks for testing agents based on this projects and fixes two minor issues that surfaced during development.

**Sanity checked all tasks pass with the sage agent:**
<img width="1510" height="947" alt="Screenshot 2026-04-26 at 14 58 48" src="https://github.com/user-attachments/assets/b79fcf2e-d6fd-4954-b4dc-385b193dd387" />

**Running the tasks with Claude:** 
Performance varied a bit across a few different runs. On the most recent ones, Claude sometimes forgot to do a dbt run after generating the solution missing out on valuable error info and it particularly struggled with recreating json parsing / un-nesting functionality.
<img width="1488" height="948" alt="Screenshot 2026-04-26 at 15 12 08" src="https://github.com/user-attachments/assets/860f275e-172b-41ea-af19-1da3e9ef81e4" />

bstancil commented:
hey @MajaCernja , welcome! This would be great to include, if you're got ideas for generating data or for tasks. (I'm also working on generated data for a larger dbt project, so if I end up creating a programmatic way to do that, I can share it with you)

MajaCernja committed:
Add slack analytics task 1 recreating fct model

Removing and recreating fct model from scratch

MajaCernja committed:
Add slack analytics task 2 recreating custom macro

Remove and recreate custom macro for unnesting json

MajaCernja committed:
Add slack analytics task 3 to fix type bug

Introduce a type mismatch bug to be fixed by the agent

MajaCernja committed:
Add slack analytics task 4 solving incorrect CTE reference

Rewrites a column ref in fct for agent to identify correct replacement

MajaCernja committed:
Propagate built-in TimeoutError

When send_blocking_keys raises the builtin TimeoutError (a subclass of OSError) it  gets misclassified as UNKNOWN_AGENT_ERROR instead of a timeout.

MajaCernja committed:
Fix JSON <> VARCHAR duckdb comparison issue

DuckDB treats JSON and VARCHAR as incompatible types, so agents that produced correct data but used a different unnesting approach would fail the equality test with a type mismatch error leading to a false negative result.

MajaCernja committed:
Add slack analytics task 5 fix dedupe bug

The setup script inserts a newer extraction of an existing message, the agent must ensure the model does not contain stale duplicate data.

MajaCernja committed:
Add slack analytics task 6 build mart model

This task requires an agent to create a mart model from scratch to create an overview of messaging + reaction activity.

MajaCernja committed:
Add slack analytics dbt project

Internal analytics project adapted to be ok for open source sharing.

MajaCernja committed:
Add helper script to mock slack reactions data

Generates reaction data which was difficult to create well with data mocking online tools.

MajaCernja commented on scripts_python/generate_channel_messages_reactions.py:
Not sure if it's all that useful to commit this script given it served quite a niche purpose, but maybe it'd be helpful as an example for future mock data generation. 

(The rest of the data I generated with an online tool called Mockaroo.)
